api/v1beta1/clusteripamclaim_types.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2024
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"errors"
+	"net"
+	"net/netip"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ClusterIPAMClaimKind Denotes the clusteripamclaim resource Kind.
+	ClusterIPAMClaimKind = "ClusterIPAMClaim"
+
+	// InvalidClaimConditionType Denotes that the claim is invalid
+	InvalidClaimConditionType = "InvalidClaimCondition"
+)
+
+// ClusterIPAMClaimSpec defines the desired state of ClusterIPAMClaim
+type ClusterIPAMClaimSpec struct {
+	// The provider that this claim will be consumed by
+	Provider string `json:"provider,omitempty"`
```

Comment:
should we restrict available options to some enum, making the current names (and renaming to `in-cluster`, `infoblox`) available publicly?

Reply:
It would be perfect to have the constants reflecting in the `// +kubebuilder:validation:Enum=in-cluster;ipam-infoblox` to be defined in the public API package, just in case